test/test_grocy.py

```diff
@@ -93,49 +86,47 @@ def test_get_shopping_list_valid(self):
     @responses.activate
     def test_get_shopping_list_invalid_no_data(self):
         responses.add(responses.GET, '{}:{}'.format(CONST_BASE_URL,CONST_PORT) + "/api/objects/shopping_list", status=400)
-        assert self.grocy.shopping_list() is None
+        self.assertRaises(HTTPError, self.grocy.shopping_list)
         
     @responses.activate
     def test_get_shopping_list_invalid_missing_data(self):
-        resp = [
-            {
-            }
-        ]
+        resp = []
         responses.add(responses.GET, '{}:{}'.format(CONST_BASE_URL,CONST_PORT) + "/api/objects/shopping_list", json=resp, status=200)
+        assert len(self.grocy.shopping_list()) == 0
         
     def test_add_missing_product_to_shopping_list_valid(self):
-        assert self.grocy.add_missing_product_to_shopping_list().status_code == 204
+        assert not self.grocy.add_missing_product_to_shopping_list()
```

Comment:
these assertions seem weird to me but I guess this i python related

Reply:
Because when it's ok, nothing is returned (204 status)
And raise an http error on error.